images/pyspark-notebook/Dockerfile

```diff
@@ -63,7 +63,7 @@ USER ${NB_UID}
 RUN mamba install --yes \
     'grpcio-status' \
     'grpcio' \
-    'pandas=2.0.3' \
+    'pandas=2.2.1' \
```

Comment:
Have you checked the comment above to make sure you’re using the proper version?

Reply:
They have `'pandas<=2.2.1` so we should be good here.

Reply:
https://github.com/apache/spark/blob/master/dev/infra/Dockerfile

Reply:
You need to check the latest stable tag, not the current main branch.

Reply:
Ah sorry there. Looks like we'll need for their next release, as this commit isn't included in the latest stable tag: https://github.com/ericm-db/spark/commit/98ca3ea2c31ce32576aab52ccfe62ab7c6e06f0b

Reply:
I think we'll need to wait on this as well, since we need at least Pandas 2.1.1 to ensure compatibility with 3.12.

Reply:
@bjornjorgensen could you please tell us when the Spark release will include this commit? (at least approximately)

Reply:
hmm.. well we are waiting for hadoop 3.4.0 and a new hive release. We haven't started any RC release yet. I build and test my own jupyterlab https://github.com/bjornjorgensen/jupyter-spark-master-docker and I did try python 3.12 but it breake so match so I'm using python 3.11 as debian testing are using. 
And Spark 3.5.1 don't support python 3.12 have a look at https://github.com/apache/spark/pull/43922

Reply:
Could we just revert this particular change rather than waiting for the next release of Spark? This is an incredibly self-imposed blocker.

Reply:
They have pinned version 2.2.2 https://github.com/apache/spark/commit/356830ada6c6ebbf54e7852c37266c32bfa137ea